dotCMS/src/main/java/com/dotcms/cluster/business/ServerAPIImpl.java

```diff
@@ -35,52 +44,111 @@ public void saveServer(Server server) throws DotDataException {
 	public Server getServer(String serverId) {
 		return serverFactory.getServer(serverId);
 	}
+	@Override
+    public Server getOrCreateServer(final String serverId) throws DotDataException {
+	    Server tryServer = getServer(serverId);
+        if(tryServer==null || tryServer.getServerId() ==null)  {
+            LocalTransaction.wrap(() ->{
+                Server newServer = new Server();
+                newServer.setServerId(serverId);
+                newServer.setIpAddress(ClusterFactory.getIPAdress());
+    
+                String hostName = "localhost";
+                try {
+                    hostName = InetAddress.getLocalHost().getHostName();
+    
+                } catch (UnknownHostException e) {
+                    Logger.error(ClusterFactory.class, "Error trying to get the host name. ", e);
+                }
+    
+                newServer.setName(hostName);
+    
+                saveServer(newServer);
+    
+                // set up ports
+    
+                String port=ClusterFactory.getNextAvailablePort(newServer.getServerId(), ServerPort.CACHE_PORT);
+                Config.setProperty(ServerPort.CACHE_PORT.getPropertyName(), port);
+                newServer.setCachePort(Integer.parseInt(port));
+    
+                port=new ESClient().getNextAvailableESPort(newServer.getServerId(), newServer.getIpAddress(), String.valueOf(newServer.getEsTransportTcpPort()));
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Split this 161 characters long line (which is greater than 150 authorized). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00103)